### PR TITLE
Remove ancillary pages and corresponding menu nav items

### PR DESCRIPTION
**Pull request by crandmck**
Per discussion in TC and:
- https://github.com/expressjs/discussions/issues/211
- https://github.com/expressjs/discussions/issues/105

Details are laid out in https://github.com/expressjs/expressjs.com/wiki/Discuss:-Remove-pages-from-docs.

Not previously discussed: The home page has a "Frameworks" tile with a link to one of the pages being removed.
- For the English page, I replaced this with   
```
      <h3>Middleware</h3> 
      Express is a routing framework with minimal functionality of its own.  A lot of
      functionality is contained in Express <a href="{{ page.lang }}/resources/middleware.html">middleware</a> modules.
```
- For non-en pages, I removed the text with no replacement.

### CODE REVIEW

**Review by wesleytodd — APPROVED** [review]

File: index.md [hunk 1]
```diff
@@ -35,8 +35,10 @@ redirect_from: "/en/index.html"
       <h3>Performance</h3> Express provides a thin layer of fundamental web application features, without obscuring Node.js features that you know and love.
     </div>
 
-    <div id="frameworks">
-      <h3>Frameworks</h3> Many <a href="{{ page.lang }}/resources/frameworks.html">popular frameworks</a> are based on Express.
+    <div id="middleware">
+      <h3>Middleware</h3> 
+      Express is a routing framework with minimal functionality of its own.  A lot of
+      functionality is contained in Express <a href="{{ page.lang }}/resources/middleware.html">middleware</a> modules.
```
Comment (jonchurch) [inline index.md]:
```suggestion
      Express is a lightweight and flexible routing framework that provides a minimal core feature set.
      Its capabilities are designed to be augmented through the use of Express <a href="{{ page.lang }}/resources/middleware.html">middleware</a> modules.
```

What do you think about this? I'd rather we state "minimal feature set" vs "minimal functionality"

**Review by jonchurch — CHANGES_REQUESTED** [review]
Read through the english removals. We are nuking some whole pages, but also removing references to those pages from other content, which is why the diff is as large as it is.

I left a suggestion for wording, LGTM either way though

File: index.md (hunk 1 above)
Comment (crandmck) [reply index.md]: Thanks @jonchurch I agree "minimal functionality" does seem rather negative.  

Your suggestion is good, but on reflection, I feel we should be even more succinct, and just say something like 

```
Express is a lightweight and flexible routing framework with minimal core features 
that is meant to be augmented through the use of Express middleware modules (link).
```

If you like that, I'll amend the PR.

TBH, my original proposal was focussed on removing the pages with links to 3p sites, and I hadn't thought through the changes required to the home page.  But of course, it's of particular importance, so I appreciate your attention to the matter.

Comment (jonchurch) [reply index.md]: Love it ty rand, LGTM! (with the typo correction of "features that is" to "features that are")

Going to approve as is, bc we can always take a follow up PR if you don't get around to landing your newest suggestion.

**Review by jonchurch — APPROVED** [review]

**Review by UlisesGascon — APPROVED** [review]

**Pull request merged**